benchmark/run.js

```diff
@@ -63,42 +64,52 @@ if (format === 'csv') {
     );
   }
 
-  if (format !== 'csv') {
-    console.log();
-    console.log(filename);
-  }
-
-  child.on('message', (data) => {
-    if (data.type !== 'report') {
-      return;
-    }
-    // Construct configuration string, " A=a, B=b, ..."
-    let conf = '';
-    for (const key of Object.keys(data.conf)) {
-      if (conf !== '')
-        conf += ' ';
-      conf += `${key}=${JSON.stringify(data.conf[key])}`;
-    }
-    if (format === 'csv') {
-      // Escape quotes (") for correct csv formatting
-      conf = conf.replace(/"/g, '""');
-      console.log(`"${data.name}", "${conf}", ${data.rate}, ${data.time}`);
-    } else {
-      let rate = data.rate.toString().split('.');
-      rate[0] = rate[0].replace(/(\d)(?=(?:\d\d\d)+(?!\d))/g, '$1,');
-      rate = (rate[1] ? rate.join('.') : rate[0]);
-      console.log(`${data.name} ${conf}: ${rate}`);
-    }
+  return new Promise((resolve, reject) => {
+    child.on('message', (data) => {
+      if (data.type !== 'report') {
+        return;
+      }
+      // Construct configuration string, " A=a, B=b, ..."
+      let conf = '';
+      for (const key of Object.keys(data.conf)) {
+        if (conf !== '')
+          conf += ' ';
+        conf += `${key}=${JSON.stringify(data.conf[key])}`;
+      }
+      if (format === 'csv') {
+        // Escape quotes (") for correct csv formatting
+        conf = conf.replace(/"/g, '""');
+        console.log(`"${data.name}", "${conf}", ${data.rate}, ${data.time}`);
+      } else {
+        let rate = data.rate.toString().split('.');
+        rate[0] = rate[0].replace(/(\d)(?=(?:\d\d\d)+(?!\d))/g, '$1,');
+        rate = (rate[1] ? rate.join('.') : rate[0]);
+        console.log(`${data.name} ${conf}: ${rate}`);
+      }
+    });
+    child.once('close', (code) => {
+      if (code) {
+        reject(code);
+      } else {
+        resolve(code);
+      }
+    });
   });
+}
 
-  child.once('close', (code) => {
-    if (code) {
-      process.exit(code);
+async function run() {
+  for (let i = 0; i < benchmarks.length; ++i) {
+    let runs = cli.optional.runs ?? 1;
+    const filename = benchmarks[i];
+    if (format !== 'csv') {
+      console.log();
+      console.log(filename);
     }
 
-    // If there are more benchmarks execute the next
-    if (i + 1 < benchmarks.length) {
-      recursive(i + 1);
+    while (runs-- > 0) {
+      await runBenchmark(filename);
     }
-  });
-})(0);
+  }
+}
+
+run();
```

Comment:
Should we handle a possible rejection of this fn? For any reason

Reply:
It's not necessary, it should not reject and if reject tests will fail.